python/lsst/ap/association/mpSkyEphemerisQuery.py

```diff
@@ -156,8 +156,9 @@ def run(self, predictedRegionTime):
         mpSkySsObjects, elements = self._mpSkyConeSearch(expCenter, expMidPointEPOCH,
                                                          expRadius + self.config.queryBufferRadiusDegrees,
                                                          mpSkyURL)
-        for element in ELEMENTS_LIST:
-            mpSkySsObjects[self.config.mpcorb_prefix + NAMING_MAP[element]] = elements[element]
+        if elements is not None:
+            for element in ELEMENTS_LIST:
+                mpSkySsObjects[self.config.mpcorb_prefix + NAMING_MAP[element]] = elements[element]
```

Comment:
This works, but it would be more robust to guarantee that `_mpSkyConeSearch` always returns a mapping (I assume? It's not documented...). That way, you don't need a separate check for every future use of `elements`.

A simple way to enforce the guarantee is to have `if elements is None: elements = {}` in the request parsing code.

Reply:
Out of scope for this work item, but it seems a bit strange that `_mpSkyConeSearch` returns two "pieces" and then `run` combines them. Is there some deeper significance I'm missing?